app/controllers/api/cloud_templates_controller.rb

```diff
@@ -11,7 +11,10 @@ def import_resource(_type, _id, data = {})
 
       ems_dst   = resource_search(data['dst_provider_id'], :providers)
       ems_src   = resource_search(data['src_provider_id'], :providers)
-      src_image = resource_search(data['src_image_id'],    :providers, collection_class(:templates))
+      src_image = resource_search(data['src_image_id'], :providers, collection_class(:templates))
```

Comment:
Does it simplify things if we can rewrite the `src_image` lookup to filter by ems (which I assume is the RBAC filter on the templates in the first place)

`ems_src.templates.find(data['src_image_id'])` -- not sure if there are other ways to filter templates for RBAC, but that would allow us to drop the later guard clause.

Not a fan of looking up the template, and them giving out too much information about the template. Though with this PR, at least the template is visible to the end user. So not too much information leaking

Reply:
Wouldn't doing so bypass an RBAC check for the template by the id of `data['src_image_id`']?